sklearn/linear_model/ridge.py

```diff
@@ -1025,7 +1032,14 @@ def _values(self, alpha, y, v, Q, QT_y):
 
     def _pre_compute_svd(self, X, y, centered_kernel=True):
         if sparse.issparse(X):
-            raise TypeError("SVD not supported for sparse matrices")
+            try:
+                X = X.toarray()
+            except MemoryError:
+                msg = ("Setting gcv_mode='svd' with a sparse X creates a "
+                       "n_samples * n_features dense matrix, setting "
+                       "gcv_mode='eigen' may help because it would create a "
+                       "n_samples**2 dense matrix")
+                raise MemoryError(msg)
```

Comment:
Maybe use a `raise ... from` since 0.21 is `python` 3 only? :)

Reply:
Do you happen to use the `__cause__` attribute of an exception? The major different from using and not using `from` is:

**from** traceback
```
MemoryError: ...

The above exception was the direct cause of the following exception:

MemoryError: ...
```

**no from** traceback
```
MemoryError: ...

During handling of the above exception, another exception occurred:

MemoryError: ...
```

Added `raise ... from`. :)

Reply:
Indeed, I tested it myself locally and the change is very subtle :). I did not know this syntax until recently and it seemed like the right use-case. 

> Do you happen to use the __cause__ attribute of an exception?

One day, I wish...